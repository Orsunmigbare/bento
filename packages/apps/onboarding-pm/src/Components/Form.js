import React from "react"
import styled from "styled-components"
import { Wrapper } from "@staccx/base"
import { spacing } from "@staccx/theme"

const Form = ({ children }) => (
  <FormWrapper size="small">{children}</FormWrapper>
)

const FormWrapper = styled(Wrapper)`
  padding-top: ${spacing.large};
  display: grid;
  grid-template-columns: 1fr;
  grid-row-gap: 48px;
`

export default Form
