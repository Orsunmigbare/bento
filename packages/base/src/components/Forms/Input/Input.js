/**
 * @class MaskedInput
 */

import React from "react"
import PropTypes from "prop-types"
import MaskedInput from "./MaskedInput"
import styled, { css } from "styled-components"
import Label from "../Label/Label"

class Input extends React.Component {
  constructor(props) {
    super(props)
    this.focus = this.focus.bind(this)
  }

  focus() {
    this.input.focus()
  }

  render() {
    const {
      autoFocus,
      className,
      disabled,
      id,
      name,
      onBlur,
      onChange,
      onKeyDown,
      placeholder,
      type,
      value,
      label,
      mask,
      ...otherProps
    } = this.props

    return (
      <InputWrapper className={className}>
        <Label htmlFor={id}>{label}</Label>
        {mask ? (
          <InputWithMask
            mask={mask}
            autoFocus={autoFocus}
            value={value}
            disabled={disabled}
            id={id}
            name={name}
            onBlur={onBlur}
            onChange={onChange}
            onKeyDown={onKeyDown}
            placeholder={placeholder}
            type={type}
            {...otherProps}
          />
        ) : (
          <InputNoMask
            autoFocus={autoFocus}
            value={value}
            disabled={disabled}
            id={id}
            name={name}
            onBlur={onBlur}
            onChange={onChange}
            onKeyDown={onKeyDown}
            placeholder={placeholder}
            type={type}
            {...otherProps}
          />
        )}
      </InputWrapper>
    )
  }
}

export const InputWrapper = styled.div`
  display: block;
  margin-bottom: 0;
  position: relative;
`

export const inputCss = css`
  display: block;
  width: 100%;
  margin: 0 auto;
  min-height: ${p => p.theme.targetSize.normal};
  border-width: 0;
  border-radius: ${p => p.theme.borderRadius};
  padding-left: ${p => p.theme.spacing.small};
  padding-right: ${p => p.theme.spacing.small};
  font-family: ${p => p.theme.font.body};
  transition: border-color 0.2s ease-out;
  font-size: ${p => p.theme.font.size.input};
  -webkit-appearance: none;
  appearance: none;
  -moz-appearance: textfield;
  border: 1px solid ${p => p.theme.color.line};

  &::-webkit-input-placeholder {
    /* WebKit browsers */
    color: ${p => p.theme.color.line};
  }
  &:-moz-placeholder {
    /* Mozilla Firefox 4 to 18 */
    color: ${p => p.theme.color.line};
  }
  &::-moz-placeholder {
    /* Mozilla Firefox 19+ */
    color: ${p => p.theme.color.line};
  }
  &:-ms-input-placeholder {
    /* Internet Explorer 10+ */
    color: ${p => p.theme.color.line};
  }
  &:-webkit-autofill {
    background-color: ${p => p.theme.color.bg};
  }

  &:hover,
  &:focus,
  &:active {
    outline: none;
    background-color: ${p => p.theme.color.subtleHover};
  }

  &::-webkit-inner-spin-button,
  &::-webkit-outer-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }
`

const InputWithMask = styled(MaskedInput)`
  ${inputCss};
`

const InputNoMask = styled.input`
  ${inputCss};
`

export const InputPropTypes = {
  autoFocus: PropTypes.bool,
  callback: PropTypes.func,
  disabled: PropTypes.bool,
  hidden: PropTypes.bool,
  id: PropTypes.string.isRequired,
  name: PropTypes.string,
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  onKeyDown: PropTypes.func,
  placeholder: PropTypes.string,
  style: PropTypes.object,
  type: PropTypes.oneOf(["text", "email", "number", "tel", "url"]),
  value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  label: PropTypes.string
}

Input.propTypes = InputPropTypes

export const InputDefaultProps = {
  autoFocus: null,
  callback: null,
  disabled: false,
  hidden: false,
  name: null,
  onBlur: null,
  onChange: null,
  onKeyDown: null,
  placeholder: null,
  style: null,
  type: "text",
  value: undefined,
  label: ""
}

Input.defaultProps = InputDefaultProps
/** @component */
export default Input
