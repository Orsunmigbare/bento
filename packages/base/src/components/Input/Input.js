/**
 * @class MaskedInput
 */

import React from "react"
import PropTypes from "prop-types"
import MaskedInput from "./MaskedInput"
import styled, { css } from "styled-components"
import Label from "../Label/Label"

class Input extends React.Component {
  constructor(props) {
    super(props)
    this.focus = this.focus.bind(this)
  }

  focus() {
    this.input.focus()
  }

  render() {
    const {
      autoFocus,
      className,
      disabled,
      id,
      name,
      onBlur,
      onChange,
      onKeyDown,
      placeholder,
      type,
      value,
      label,
      mask,
      ...otherProps
    } = this.props

    return (
      <InputWrapper className={className}>
        <Label htmlFor={id}>{label}</Label>
        {mask ? (
          <InputWithMask
            mask={mask}
            autoFocus={autoFocus}
            value={value}
            disabled={disabled}
            id={id}
            name={name}
            onBlur={onBlur}
            onChange={onChange}
            onKeyDown={onKeyDown}
            placeholder={placeholder}
            type={type}
            {...otherProps}
          />
        ) : (
            <InputNoMask
              autoFocus={autoFocus}
              defaultValue={value}
              disabled={disabled}
              id={id}
              name={name}
              onBlur={onBlur}
              onChange={onChange}
              onKeyDown={onKeyDown}
              placeholder={placeholder}
              type={type}
              {...otherProps}
            />
          )}
      </InputWrapper>
    )
  }
}

export const InputWrapper = styled.div`
  display: block;
  margin-bottom: 0;
  position: relative;
`

export const inputCss = css`
display: block;
  width: 100%;
  margin: 0 auto;
  min-height: 72px;
  border-width: 0;
  border-radius: 0;
  padding: ${p => p.theme.spacing.medium} ${p => p.theme.spacing.medium} 0;
  font-family: ${p => p.theme.font.body};
  transition: border-color .2s ease-out;
  font-size: ${p => p.theme.font.size.input};
  -webkit-appearance: none;
  appearance: none;
  -moz-appearance: textfield;


  &::-webkit-input-placeholder { /* WebKit browsers */
    color: ${p => p.theme.color.line};
  }
  &:-moz-placeholder { /* Mozilla Firefox 4 to 18 */
    color: ${p => p.theme.color.line};
  }
  &::-moz-placeholder { /* Mozilla Firefox 19+ */
    color: ${p => p.theme.color.line};
  }
  &:-ms-input-placeholder { /* Internet Explorer 10+ */
    color: ${p => p.theme.color.line};
  }
  &:-webkit-autofill {
    background-color: ${p => p.theme.color.bg};
  }

  &:hover,
  &:focus,
  &:active {
    outline: none;
    border-width: 0;
    background-color: ${p => p.theme.color.subtleHover};
  }

  &::-webkit-inner-spin-button,
  &::-webkit-outer-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }
`

const InputWithMask = styled(MaskedInput) `
  ${inputCss};
`

const InputNoMask = styled.input`
  ${inputCss};
`

export const InputPropTypes = {
  autoFocus: PropTypes.bool,
  callback: PropTypes.func,
  disabled: PropTypes.bool,
  hidden: PropTypes.bool,
  id: PropTypes.string.isRequired,
  name: PropTypes.string,
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  onKeyDown: PropTypes.func,
  placeholder: PropTypes.string,
  style: PropTypes.object,
  type: PropTypes.oneOf(["text", "email", "number", "tel", "url"]),
  value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  label: PropTypes.string
}

Input.propTypes = InputPropTypes

export const InputDefaultProps = {
  autoFocus: null,
  callback: null,
  disabled: false,
  hidden: false,
  name: null,
  onBlur: null,
  onChange: null,
  onKeyDown: null,
  placeholder: null,
  style: null,
  type: "text",
  value: "",
  label: ""
}

Input.defaultProps = InputDefaultProps
/** @component */
export default Input
