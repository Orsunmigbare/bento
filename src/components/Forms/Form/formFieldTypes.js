export default {
  string: "string",
  number: "number",
  bool: "bool",
  email: "email",
  date: "date",
  url: "url"
}
