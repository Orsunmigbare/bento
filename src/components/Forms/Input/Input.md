# Input
Common input for texts. Use by other components and is a forward ref component.
This means you can send it a ref and you get your ref.current as the input (to do focus() and blur() and so on)