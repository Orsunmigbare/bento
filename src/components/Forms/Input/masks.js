export const accountMasks = {
  nb: { blocks: [4, 2, 5], numericOnly: true, delimiter: " " }
}
