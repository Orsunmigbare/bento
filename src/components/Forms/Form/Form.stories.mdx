import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import Form from "./Form"
import FormField from "./FormField"

<Meta title="Components|Forms/Form" component={Form}/>


# Form
Forklaring på hva det er og hvordan den kan brukes


# import

```javascript
import {Form} from "@staccx/bento"
```

# Types of Forms


## Standard Form
This is the standard Form

<Preview>
  <Story name="Standard">
       <Form initialValues={{ epost: "" }} buttonText="Søk nå">
             <FormField
               name="epost"
               type="email"
               required
               renderError={({ error, touched }) => (
                 <span>
                   {error} {touched}
                 </span>
               )}
             >
               {({ name, field, form }) => (
                 <Input {...field} placeholder="ola@nordmann.no" id={name} />
               )}
             </FormField>
           </Form>
  </Story>
</Preview>


### Source code
<details><summary>Form</summary>
<p>

```bash
import React, { Component } from "react"
import PropTypes from "prop-types"
import { Formik } from "formik"
import Button from "../../Button/Button"
import { commonPropTypes } from "../../../theming/themeContants"
import traverse from "react-traverse"
const Yup = require("yup")

const findFormFields = (node, startFields = []) => {
  const formFields = startFields
  const isFormField = name => name === "FormField"

  traverse(node, {
    ComponentElement(path) {
      if (isFormField(path.node.type.name)) {
        formFields.push(path.node)
      }
      path.traverseChildren()
    }
  })

  return formFields
}

class Form extends Component {
  render() {
    const formFields = findFormFields(this.props.children)
    const validationShape = formFields.reduce((validator, curr) => {
      const { type, required, nullable, name, min, max, integer } = curr.props

      switch (type) {
        case "string": {
          validator[name] = Yup.string()
          break
        }
        case "number": {
          validator[name] = Yup.number()
          if (min) {
            validator[name] = validator[name].min(min)
          }
          if (max) {
            validator[name] = validator[name].max(max)
          }
          if (integer) {
            validator[name] = validator[name].integer()
          }
          break
        }
        case "bool": {
          validator[name] = Yup.bool()
          break
        }
        case "email": {
          validator[name] = Yup.string().email()
          break
        }
        case "date": {
          validator[name] = Yup.date()
          break
        }
        case "url": {
          validator[name] = Yup.string().url()
          break
        }
        default: {
          validator[name] = Yup.string()
          break
        }
      }
      if (nullable) {
        validator[name] = validator[name].nullable()
      }
      if (required) {
        validator[name] = validator[name].required("Denne må du ha")
      }
      return validator
    }, {})
    const validationSchema = Yup.object().shape(validationShape)
    const { Container } = this.props
    return (
      <Formik
        initialValues={this.props.initialValues}
        validationSchema={validationSchema}
        onSubmit={this.props.onSubmit}
        render={({ handleSubmit, values, isValid, errors, touched }) => {
          return (
            <form onSubmit={handleSubmit}>
              {Container && (
                <Container {...this.props.containerProps}>
                  {this.props.children}
                  {this.props.renderButton ? (
                    this.props.renderButton({
                      values,
                      isValid,
                      errors,
                      touched
                    })
                  ) : (
                    <Button type="submit" onClick={() => null}>
                      {this.props.buttonText}
                    </Button>
                  )}
                </Container>
              )}
              {!Container && (
                <>
                  {this.props.children}
                  {this.props.renderButton ? (
                    this.props.renderButton({ values, isValid, errors })
                  ) : (
                    <Button type="submit" onClick={() => null}>
                      {this.props.buttonText}
                    </Button>
                  )}
                </>
              )}
            </form>
          )
        }}
      />
    )
  }
}

Form.propTypes = {
  children: commonPropTypes.children.isRequired,
  initialValues: PropTypes.object,
  onSubmit: PropTypes.func,
  renderButton: PropTypes.func
}

Form.defaultProps = {
  initialValues: {},
  onSubmit: console.log,
  containerProps: {}
}

export default Form


```

</p>
</details>



# Props for Form
<Props of={Form} />
