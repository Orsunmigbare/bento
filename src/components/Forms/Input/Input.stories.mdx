import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import Input from "./Input"

<Meta title="Components|Forms/Input/Input" component={Input}/>


# Input


```javascript
import {Input} from "@staccx/bento"
```

## Standard Input

<Preview>
  <Story name="Standard">
   <Input label={"Input"} />
  </Story>
</Preview>


## AutoFocus
With props autoFocus, will set focus to the input field when the page renders

<Preview>
  <Story name="AutoFocus">
   <Input label={"Input"}  autoFocus/>
  </Story>
</Preview>

## Disabled
With disabled prop, Input field can not be used

<Preview>
  <Story name="Disabled">
   <Input label={"Input"}  disabled/>
  </Story>
</Preview>


## Props for Input
<Props of={Input} />
