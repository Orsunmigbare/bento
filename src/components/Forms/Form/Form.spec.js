import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../theming/themes/baseTheme"
import { render, screen } from "@testing-library/react"
import Form from "./Form"
import FormField from "./FormField"
import Input from "../Input/Input"

describe("Form", () => {
  describe("Snapshots", () => {
    it("Email", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Form initialValues={{ epost: "" }} buttonText="Søk nå">
              <FormField
                name="epost"
                type="email"
                required
                renderError={({ error, touched }) => (
                  <span>
                    {error} {touched}
                  </span>
                )}
              >
                {({ name, field, form }) => (
                  <Input {...field} placeholder="ola@nordmann.no" id={name} />
                )}
              </FormField>
            </Form>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("Form", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Form initialValues={{ epost: "" }} buttonText="Søk nå">
              <FormField
                name="epost"
                required
                renderError={({ error, touched }) => (
                  <span>
                    {error} {touched}
                  </span>
                )}
              >
                {({ name, field, form }) => (
                  <Input {...field} placeholder="ola@nordmann.no" id={name} />
                )}
              </FormField>
            </Form>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("Form", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Form initialValues={{ epost: "" }} buttonText="Søk nå">
              <FormField
                name="epost"
                type="date"
                required
                renderError={({ error, touched }) => (
                  <span>
                    {error} {touched}
                  </span>
                )}
              >
                {({ name, field, form }) => (
                  <Input {...field} placeholder="ola@nordmann.no" id={name} />
                )}
              </FormField>
            </Form>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("url", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Form initialValues={{ epost: "" }} buttonText="Søk nå">
              <FormField
                name="epost"
                type="url"
                required
                renderError={({ error, touched }) => (
                  <span>
                    {error} {touched}
                  </span>
                )}
              >
                {({ name, field, form }) => (
                  <Input {...field} placeholder="ola@nordmann.no" id={name} />
                )}
              </FormField>
            </Form>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("bool", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Form initialValues={{ epost: "" }} buttonText="Søk nå">
              <FormField
                name="epost"
                type="bool"
                required
                renderError={({ error, touched }) => (
                  <span>
                    {error} {touched}
                  </span>
                )}
              >
                {({ name, field, form }) => (
                  <Input {...field} placeholder="ola@nordmann.no" id={name} />
                )}
              </FormField>
            </Form>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("number", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Form initialValues={{ epost: "" }} buttonText="Søk nå">
              <FormField
                name="epost"
                type="number"
                required
                renderError={({ error, touched }) => (
                  <span>
                    {error} {touched}
                  </span>
                )}
              >
                {({ name, field, form }) => (
                  <Input {...field} placeholder="ola@nordmann.no" id={name} />
                )}
              </FormField>
            </Form>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("string", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Form initialValues={{ epost: "" }} buttonText="Søk nå">
              <FormField
                name="epost"
                type="string"
                min={0}
                max={100}
                integer
                required
                renderError={({ error, touched }) => (
                  <span>
                    {error} {touched}
                  </span>
                )}
              >
                {({ name, field, form }) => (
                  <Input {...field} placeholder="ola@nordmann.no" id={name} />
                )}
              </FormField>
            </Form>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})

describe("Rendering", () => {
  it("Should render", () => {
    render(
      <ThemeProvider theme={baseTheme}>
        <Form initialValues={{ epost: "" }} buttonText="Søk nå">
          <FormField
            name="epost"
            type="string"
            min={0}
            max={100}
            integer
            required
            renderError={({ error, touched }) => (
              <span>
                {error} {touched}
              </span>
            )}
          >
            {({ name, field, form }) => (
              <Input {...field} placeholder="ola@nordmann.no" id={name} />
            )}
          </FormField>
        </Form>
      </ThemeProvider>
    )
    expect(screen.getByPlaceholderText("ola@nordmann.no")).toBeInTheDocument()
  })
})
