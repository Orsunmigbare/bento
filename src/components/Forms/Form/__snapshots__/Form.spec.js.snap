// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Snapshots Email 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="sc-fzpans hIlNMX"
  >
    
    
    <input
      autoFocus={null}
      className="sc-fzplWN kUtfRN"
      disabled={false}
      hidden={false}
      id="epost"
      name="epost"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={null}
      placeholder="ola@nordmann.no"
      style={null}
      type="text"
      value=""
    />
  </div>
  <span>
     
  </span>
  <button
    className="sc-AxjAm ecAfij"
    onClick={[Function]}
    type="submit"
  >
    Søk nå
  </button>
</form>
`;

exports[`Form Snapshots Form 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="sc-fzpans hIlNMX"
  >
    
    
    <input
      autoFocus={null}
      className="sc-fzplWN kUtfRN"
      disabled={false}
      hidden={false}
      id="epost"
      name="epost"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={null}
      placeholder="ola@nordmann.no"
      style={null}
      type="text"
      value=""
    />
  </div>
  <span>
     
  </span>
  <button
    className="sc-AxjAm ecAfij"
    onClick={[Function]}
    type="submit"
  >
    Søk nå
  </button>
</form>
`;

exports[`Form Snapshots Form 2`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="sc-fzpans hIlNMX"
  >
    
    
    <input
      autoFocus={null}
      className="sc-fzplWN kUtfRN"
      disabled={false}
      hidden={false}
      id="epost"
      name="epost"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={null}
      placeholder="ola@nordmann.no"
      style={null}
      type="text"
      value=""
    />
  </div>
  <span>
     
  </span>
  <button
    className="sc-AxjAm ecAfij"
    onClick={[Function]}
    type="submit"
  >
    Søk nå
  </button>
</form>
`;

exports[`Form Snapshots bool 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="sc-fzpans hIlNMX"
  >
    
    
    <input
      autoFocus={null}
      className="sc-fzplWN kUtfRN"
      disabled={false}
      hidden={false}
      id="epost"
      name="epost"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={null}
      placeholder="ola@nordmann.no"
      style={null}
      type="text"
      value=""
    />
  </div>
  <span>
     
  </span>
  <button
    className="sc-AxjAm ecAfij"
    onClick={[Function]}
    type="submit"
  >
    Søk nå
  </button>
</form>
`;

exports[`Form Snapshots number 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="sc-fzpans hIlNMX"
  >
    
    
    <input
      autoFocus={null}
      className="sc-fzplWN kUtfRN"
      disabled={false}
      hidden={false}
      id="epost"
      name="epost"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={null}
      placeholder="ola@nordmann.no"
      style={null}
      type="text"
      value=""
    />
  </div>
  <span>
     
  </span>
  <button
    className="sc-AxjAm ecAfij"
    onClick={[Function]}
    type="submit"
  >
    Søk nå
  </button>
</form>
`;

exports[`Form Snapshots string 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="sc-fzpans hIlNMX"
  >
    
    
    <input
      autoFocus={null}
      className="sc-fzplWN kUtfRN"
      disabled={false}
      hidden={false}
      id="epost"
      name="epost"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={null}
      placeholder="ola@nordmann.no"
      style={null}
      type="text"
      value=""
    />
  </div>
  <span>
     
  </span>
  <button
    className="sc-AxjAm ecAfij"
    onClick={[Function]}
    type="submit"
  >
    Søk nå
  </button>
</form>
`;

exports[`Form Snapshots url 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="sc-fzpans hIlNMX"
  >
    
    
    <input
      autoFocus={null}
      className="sc-fzplWN kUtfRN"
      disabled={false}
      hidden={false}
      id="epost"
      name="epost"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={null}
      placeholder="ola@nordmann.no"
      style={null}
      type="text"
      value=""
    />
  </div>
  <span>
     
  </span>
  <button
    className="sc-AxjAm ecAfij"
    onClick={[Function]}
    type="submit"
  >
    Søk nå
  </button>
</form>
`;
